src/Network/WebSockets/TSSocketManager.m

```diff
@@ -117,6 +118,13 @@ - (void)ensureWebsocket
     DDLogWarn(@"Creating new websocket");
 
     // If socket is not already open or connecting, connect now.
+    //
+    // First we need to close the existing websocket, if any.
+    // The websocket methods are invoked _after_ the websocket state
+    // changes, so we may be just learning about a socket failure
+    // or close event now.
+    self.status = kSocketStatusClosed;
```

Comment:
This is a key fix, as it ensures that we clear out any old failed/closed web sockets.